modal/_serialization.py

```diff
@@ -470,10 +472,23 @@ def deserialize_params(serialized_params: bytes, function_def: api_pb2.Function,
         api_pb2.ClassParameterInfo.PARAM_SERIALIZATION_FORMAT_PICKLE,
     ):
         # legacy serialization format - pickle of `(args, kwargs)` w/ support for modal object arguments
-        param_args, param_kwargs = deserialize(serialized_params, _client)
+        try:
+            param_args, param_kwargs = deserialize(serialized_params, _client)
+        except DeserializationError:
```

Comment:
Is this case we're expecting the only way we can get a `DeserializeError` here?

Reply:
There may be other ways of getting this, but in those cases the proto decode would also fail and result in container crashes afaik

Reply:
Yeah my concern is that the error message / traceback in that case will be misleading.

Reply:
I can't come up with any cases where this would be something users should have to care about, but maybe if they are pickling some custom type parameter that they don't have available in the container? Feels very niche (and also only applicable for the deprecated explicit constructor style classes)